src/middleware.ts

```diff
@@ -1,12 +1,24 @@
 import { NextResponse } from 'next/server'
 import type { NextRequest } from 'next/server'
- 
+
 // This function can be marked `async` if using `await` inside
 export async function middleware(request: NextRequest) {
+	console.log('middleware firing!')
+	// console.log(request)
+	console.log(request.cookies, Date.now())
 	try {
 		const sessionID = request.cookies.get('sessionID');
-		if (!sessionID)
+		if (!sessionID) {
+			console.log('the redirect happening?')
 			return NextResponse.redirect(new URL('/', request.url))
+		}
+
```

Comment:
over here, a fetch request to `api/admin/SessionID` will always return "No session ID found"

this is because it's a new request created by the middleware (Next.js on the server side),  
instead of a request from the browser side (that will come with the cookies attached)

Reply:
![image](https://github.com/42KL-admin/42DoGood-Reef_Check/assets/20178082/7d9265e5-bcc2-4195-8328-c5897b7ce4a3)